### PR TITLE
7568 remove raw where

### LINKED ISSUE #7568: Deprecate Interpretation of Raw Properties and Values in the Where Object
## What you are doing?
The $raw property needs to be deprecated. See https://github.com/sequelize/sequelize/blob/master/lib/dialects/abstract/query-generator.js#L2219

```js
where: {
  $raw: 'INJECTION POSSIBLE HERE'
}
```

The interpretation of a where object as literal when property value is an array also needs to be deprecated.
```js
// search with string replacements
Project.findAll({ where: ["id > ?", 25] }).then(function(projects) {
  // projects will be an array of Projects having a greater id than 25
})
```

__Dialect:__ any
__Database version:__ any
__Sequelize version:__ 4

### PR DESCRIPTION
**Pull request by mkaufmaner**
### Pull Request check-list

_Please make sure to review and check all of these items:_

- [x] Does `npm run test` or `npm run test-DIALECT` pass with this change (including linting)?
- [x] Does your issue contain a link to existing issue (Closes #[issue]) or a description of the issue you are solving?
- [x] Have you added new tests to prevent regressions?
- [x] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [x] Have you added an entry under `Future` in the changelog?

_NOTE: these things are not required to open a PR and can be done afterwards / while the PR is open._

### Description of change
Closes #7568

Remove support for literal where replacements and $raw property.

### CODE REVIEW

**Comment by mkaufmaner** [comment]
Ready for review. Just waiting on the code coverage to be reported.

File: changelog.md [hunk 1]
```diff
@@ -77,6 +78,7 @@
 - `options.order` Now only excepts values with type of array or Sequelize method.  Support for string values (ie `{order: 'name DESC'}`) has been deprecated.
 - `DataTypes.DATE` now uses `DATETIMEOFFSET` instead of `DATETIME2` sql datatype in case of MSSQL to record timezone [#5403](https://github.com/sequelize/sequelize/issues/5403)
 - `DataTypes.DECIMAL` returns string for MySQL and Postgres
+- `{ where: { $raw: "foo = 'bar'" } }` and `{ where: ['foo = ?', ['bar'] }` is no longer suported. Use `.literal` instead
```
Comment (felixfbecker) [inline changelog.md]: s/is/are

File: test/integration/instance/update.test.js [hunk 2]
```diff
@@ -430,7 +430,7 @@ describe(Support.getTestDialectTeaser('Instance'), () => {
             expect(download.finishedAt).to.not.be.ok;
 
             return Download.findAll({
-              where: dialect === 'postgres' || dialect === 'mssql' ? ['"finishedAt" IS NULL'] : ['`finishedAt` IS NULL']
+              where: dialect === 'postgres' || dialect === 'mssql' ? {finishedAt: null} : {finishedAt: null}
```
Comment (felixfbecker) [inline test/integration/instance/update.test.js]: the ternary always evaluates to the same object

File: changelog.md (hunk 1 above)
Comment (mkaufmaner) [inline changelog.md]: Wasn't me (https://www.youtube.com/watch?v=2g5Hz17C4is) but fixed.

File: test/integration/instance/update.test.js (hunk 2 above)
Comment (mkaufmaner) [inline test/integration/instance/update.test.js]: Done.

**Pull request merged**